lsm9ds1/lsm9ds1.go

```diff
@@ -61,19 +60,23 @@ func New(bus drivers.I2C) *Device {
 // Case of boolean false and error nil means I2C is up,
 // but "who am I" responses have unexpected values.
 func (d *Device) Connected() bool {
-	data1, data2 := d.buf[:1], d.buf[1:2]
-	legacy.ReadRegister(d.bus, d.AccelAddress, WHO_AM_I, data1)
-	legacy.ReadRegister(d.bus, d.MagAddress, WHO_AM_I_M, data2)
-	return data1[0] == 0x68 && data2[0] == 0x3D
+	data, err := d.readBytes(d.AccelAddress, WHO_AM_I, 1)
+	if err != nil || data[0] != 0x68 {
+		return false
+	}
+	data, err = d.readBytes(d.MagAddress, WHO_AM_I_M, 1)
```

Comment:
could this be rewritten as d.readByte (single byte return)

Reply:
Also heads up: regmap package implements read/write logic related to heapless register mapped peripherals, like this one https://github.com/tinygo-org/drivers/blob/dev/internal/regmap/devices.go

May help out in future modernizations :)